core/amber/src/main/scala/edu/uci/ics/texera/web/resource/CollaborationResource.java

```diff
@@ -0,0 +1,83 @@
+package edu.uci.ics.texera.web.resource;
+
+import edu.uci.ics.texera.web.ServletAwareConfigurator;
+
+import javax.websocket.*;
+import javax.websocket.server.ServerEndpoint;
+import java.util.HashMap;
+
+
+@ServerEndpoint(
+        value = "/automerge",
+        configurator = ServletAwareConfigurator.class
+)
+public class CollaborationResource {
+    public static HashMap<String, Session> websocketSessionMap = new HashMap<>();
+
+    public static String lockHolder = "";
+
+    // Stores state of automerge
+    public static String automergeInitial = "";
+
+    @OnOpen
+    public void myOnOpen(final Session session) {
+        websocketSessionMap.put(session.getId(), session);
+    }
+
+    @OnMessage
+    public void myOnMsg(final Session session, String message) {
+
+        message = message.substring(1, message.length() - 1);
+        // Two different things if current automergeObject is sent 
+        String temp1 = "{\"response\": \"" + message + "\"}";
+        String temp = "{" + message + "}";
+        System.out.println(temp);
+        for(String key: websocketSessionMap.keySet()) {
+            // only send to other sessions, not the session that send the message?
+            Session sess = websocketSessionMap.get(key);
+                if (sess != session) {
```

Comment:
I don't think this implementation works with the user system in the production envrionment. 

Because in the local envrionment, there's always only one workflow. 
But in the production, different users might be opening different workflows, and they will all be connecting to the same websocket resource. We should only broadcast events from the same workflow.

Reply:
Will take care of this problem in the next step.